lib/services/role.go

```diff
@@ -1794,16 +1794,22 @@ func (set RoleSet) SessionRecordingMode(service constants.SessionRecordingServic
 			// Early return as "strict" since it is the strictest value.
 			return constants.SessionRecordingModeStrict
 		case constants.SessionRecordingModeBestEffort:
-			useDefault = false
+			serviceValue = roleMode
+		case constants.SessionRecordingModeOff:
+			// "off" will only take place if the service value is unspecified.
+			// This avoids it to overwrite stricter modes.
+			if serviceValue == constants.SessionRecordingModeUnspecified {
+				serviceValue = roleMode
+			}
 		}
 	}
 
-	// Return the strictest default value.
-	if useDefault {
-		return defaultValue
+	// If service value it set, return it.
+	if serviceValue != constants.SessionRecordingModeUnspecified {
+		return serviceValue
 	}
 
-	return constants.SessionRecordingModeBestEffort
+	return defaultValue
```

Comment:
Is `off` only available to ssh? can you set `off` to default?

Reply:
> Is off only available to ssh?

`default` and `ssh`. Currently, there is only the SSH option on the `record_session` that represents the recording mode, so setting `default` or `ssh` makes no difference.

> can you set off to default?

Yes, I'll update the logic, it should follow the same as the `serviceValue`.

Reply:
Updated.